docs/getstarted/agent_evaluation.md

```diff
@@ -0,0 +1,544 @@
+
+Each time a message is exchanged during agent execution, it gets added to the messages list in the GraphState. However, Ragas requires a specific message format for evaluating interactions.
+
+Ragas uses its own format to evaluate agent interactions. So, if you're using LangGraph, you will need to convert the LangChain message objects into Ragas message objects. This allows you to evaluate your AI agents with Ragas’ built-in evaluation tools.
+
+**Goal:**  Convert the list of LangChain messages (e.g., HumanMessage, AIMessage, and ToolMessage) into the format expected by Ragas, so the evaluation framework can understand and process them properly.
+
+
+```python
+from typing import List, Union
+from langchain_core.messages import HumanMessage, SystemMessage, AIMessage, ToolMessage
+import ragas.messages as r
+import json
+
+
+def convert_to_ragas_trace(
+    messages: List[Union[HumanMessage, SystemMessage, AIMessage, ToolMessage]]
+) -> List[Union[r.HumanMessage, r.AIMessage, r.ToolMessage]]:
+    """
+    Converts a list of LangChain messages into Ragas messages for agent evaluation.
+
+    This function takes a list of LangChain message objects (which may include human messages, AI messages,
+    and tool messages) and converts them into Ragas message objects. It captures AI tool calls within each AI message,
+    if present, for more comprehensive evaluation using Ragas.
+
+    Args:
+        messages (List[Union[HumanMessage, SystemMessage, AIMessage, ToolMessage]]): A list of LangChain messages
+            representing the interaction trace of an AI agent. This may include human messages, system messages, AI messages,
+            and tool messages.
+
+    Returns:
+        List[Union[r.HumanMessage, r.AIMessage, r.ToolMessage]]: A list of Ragas message objects converted from the
+            input LangChain messages, ready for agent evaluation in Ragas.
+    """
+    ragas_trace = []
+    for message in messages:
+        if isinstance(message, HumanMessage):
+            ragas_trace.append(r.HumanMessage(content=message.content))
+        elif isinstance(message, AIMessage):
+            tool_calls = []
+            if message.additional_kwargs and "tool_calls" in message.additional_kwargs:
+                for tool_call in message.additional_kwargs["tool_calls"]:
+                    tool_calls.append(
+                        r.ToolCall(
+                            name=tool_call["function"]["name"],
+                            args=json.loads(tool_call["function"]["arguments"]),
+                        )
+                    )
+
+            if not tool_calls:
+                ragas_trace.append(r.AIMessage(content=message.content))
+            else:
+                ragas_trace.append(
+                    r.AIMessage(content=message.content, tool_calls=tool_calls)
+                )
+        elif isinstance(message, ToolMessage):
+            ragas_trace.append(r.ToolMessage(content=message.content))
+
+    return ragas_trace
```

Comment:
can you add this as part of ragas.integrations.langchain ?

Reply:
Would it be possible to add this to langgraph instead? It seems to make more sense there since it's focused on the evaluation of agents. What do you think?

Reply:
it depends on where the messages are defined in langgraph. If they import it from langchain then that is the root source, so we put our integration there. vice versa otherwise 

but ur call

Reply:
Going with ragas.integrations.langgraph.

I was thinking of using the version from the Ragas library (the one we talked about above) but keeping the current version too, so people can still take inspiration from it if they want to adapt it to their own use case.

Reply:
Do you think I should update the tutorial as well? I was thinking of using the version from the Ragas library but keeping the `convert_to_ragas_trace` as well , so people can still take inspiration from it if they want to adapt it to their own use case. What do you think?

Reply:
why did you end up choosing langgraph?

do push the latest changes too 🙂

Reply:
LangGraph can use components from LangChain to handle messages, but that doesn't mean that changes specific to LangGraph should be added to LangChain. Just because LangGraph achieves certain functions by using LangChain doesn't mean that LangChain itself should be responsible for changes that are only relevant to LangGraph.

Reply:
fair enough, making as resolved 🙂